webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/SystemVariableService.scala

```diff
@@ -530,17 +533,50 @@ class SystemVariableServiceImpl(
     val varNodeGroups = systemVariableSpecService.get("RUDDER_NODE_GROUPS_VARS").toVariable(Seq(stringNodeGroupsVars))
     val varNodeGroupsClasses = systemVariableSpecService.get("RUDDER_NODE_GROUPS_CLASSES").toVariable(Seq(stringNodeGroupsClasses))
 
-    // If Syslog is disabled, we force HTTPS
-    val varNodeReportingProtocol = getSyslogProtocolDisabled() match {
-      case Full(true)  => systemVariableSpecService.get("REPORTING_PROTOCOL").toVariable(Seq(AgentReportingHTTPS.value))
-      case Full(false) => nodeInfo.nodeReportingConfiguration.agentReportingProtocol match {
-                            case Some(protocol) => systemVariableSpecService.get("REPORTING_PROTOCOL").toVariable(Seq(protocol.value))
-                            case _              => getProp("REPORTING_PROTOCOL", () =>  getReportProtocolDefault().map(_.value))
-                          }
-      case eb:EmptyBox =>
-        val e = eb ?~! s"Failed to get information on syslog protocol global status, fallbacking to default value."
-        logger.error( e.messageChain )
-        getProp("REPORTING_PROTOCOL", () => getReportProtocolDefault().map(_.value))
+    // If Syslog is disabled, we force HTTPS.
+    val varNodeReportingProtocol = {
+      // By default, we are tolerant: it's far worse to fail a generation while support is ok than to
+      // succeed and have a node that doesn't rerport.
+      // => version must exists and and starts by 2.x, 3.x, 4.x, 5.x
+      def versionHasHttps(maybeVersion: Option[AgentVersion]): Boolean = {
```

Comment:
You could reverse the logic since you are doing looking for a False in badVersion
You could instead do a versionHasSyslogOnly
You could Put the None case to be false
and remove the not in the Some case